### PR TITLE
task/new_layers_types/display_drawer

### PR DESCRIPTION
**Pull request by BarAgam**
First part -
Refactor:
1. Moved vector-layers and rasters files into corresponding rasters and (vector)layers packages.
2. Changed raster's displaying interactor to emit notifications about rasters to be displayed with show status false (much like the vector-layers)
3. Shared code between the vector-layer and raster visibilityChanges mechanism

Second part - 
Displaying available rasters on the drawer panel.
Clicking on drawer items will effect the domain, which in turn takes care of displaying through the MapView DisplayIntermediateRasterInteractor usage

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java [hunk 1]
```diff
@@ -52,61 +65,72 @@ public void start() {
...
+                                    intermediateRasterPresentation) {
+            DrawerViewModel.this.display(
+                    intermediateRasterPresentation.getName(),
+                    intermediateRasterPresentation.getName(),
+                    intermediateRasterPresentation.isShown(),
+                    R.id.drawer_menu_submenu_rasters);
+        }
+    }
+
+    private void display(String id, String title, boolean isShown, int submenuId) {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java:99]: Move this method above the class

Comment (teamagam3) [reply app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java:99]: And the other methods below too

File: app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java [hunk 2]
```diff
@@ -11,34 +11,47 @@
 import com.teamagam.gimelgimel.app.common.base.ViewModels.BaseViewModel;
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java:11]: This class is huge.
Consider splitting it somehow

File: data/src/main/java/com/teamagam/gimelgimel/data/rasters/repository/IntermediateRasterVisibilityDataRepository.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import rx.Observable;
+
+public class IntermediateRasterVisibilityDataRepository implements IntermediateRasterVisibilityRepository {
+
+    private final ReplayRepository<IntermediateRasterVisibilityChange> mInnerRepo;
+    private String mCurrentlySelectedName;
+
+    @Inject
+    public IntermediateRasterVisibilityDataRepository() {
+        mInnerRepo = ReplayRepository.createReplayAll();
```
Comment (teamagam3) [inline data/src/main/java/com/teamagam/gimelgimel/data/rasters/repository/IntermediateRasterVisibilityDataRepository.java:20]: Why not replay(1)? Is that necessary?
If we have many events, the ViewModel will select items in order (Replay all the actions again).

SubjectFactory should be used here.

File: data/src/main/java/com/teamagam/gimelgimel/data/rasters/repository/IntermediateRasterVisibilityDataRepository.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        if (isCurrentUnselected(change)) {
+            mCurrentlySelectedName = null;
+        } else if (change.isVisible()) {
+            mCurrentlySelectedName = change.getIntermediateRasterName();
+        }
+    }
+
+    private void throwOnMultipleSelection(IntermediateRasterVisibilityChange change) {
+        if (change.isVisible() && mCurrentlySelectedName != null) {
```
Comment (teamagam3) [inline data/src/main/java/com/teamagam/gimelgimel/data/rasters/repository/IntermediateRasterVisibilityDataRepository.java:51]: Consider remove this exception, and handle it automatically here instead.

**Review by teamagam3 — APPROVED** [review]

**Comment by BarAgam** [comment]
Will be merged through rebased task right into develop.

**Pull request closed without merging**